utils/db_tracker.py

```
@@ -43,53 +48,67 @@ def ensure_tables_exist():
+                json.dumps(outcome),
+            ),
+        )
+
         conn.commit()
-        log(f"Outcome tracked in database for {protocol_name}: {outcome.get('success', 'unknown')}")
-        
+        log(
+            f"Outcome tracked in database for {protocol_name}: {
+                outcome.get(
+                    'success',
+                    'unknown')}"
```

Ruff (0.11.9)

Trailing comma missing

Add trailing comma

(COM812)